reporting/impl/src/main/java/org/jboss/windup/reporting/rules/generation/techreport/TechReportService.java

```diff
@@ -167,20 +167,21 @@ private static TagModel getNonPlaceParent(TagGraphService tagService, TagModel t
     /**
      * Prepares a precomputed matrix - map of maps of maps: rowTag -> boxTag -> project -> placement label -> TechUsageStatSum.
      *
-     * @param onlyForApplication Sum the statistics only for this project.
+     * @param appBeingSummarized Sum the statistics only for this project.
      */
-    public TechStatsMatrix getTechStatsMap(ProjectModel onlyForApplication)
+    public TechStatsMatrix getTechStatsMap(ProjectModel appBeingSummarized)
     {
-        Set<ProjectModel> applicationProjects = onlyForApplication == null ? null
-                    : new ProjectModelTraversal(onlyForApplication).getAllProjects(true);
 
         Map<String, Map<String, Map<Long, Map<String, TechUsageStatSum>>>> map = new HashMap<>();
 
```

Comment:
What do you think about refactoring this conditional ? It took me a while to understand what is filtering.

boolean appIsContainedInStatProject = stat.getProjectModel().getApplications().contains(onlyForApplication);
boolean appIsASharedApp = onlyForApplication.getName().toLowerCase().contains("shared");
boolean projectHasSeveralApps = stat.getProjectModel().getApplications().size() > 1;

if (!appIsContainedInStatProject || (pojectHasSeverlApps && !appIsASharedApp) {